### DISCUSSION TITLE
[Feature] Protected Trash

### DISCUSSION THREAD

**Discussion by chris75898** [immich-app/immich#12752] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I run Immich for myself and for family members and it is a great piece of software. Below is a feature that helps to prevent someone from accidentally deleting photos and resulting in them being lost forever. It also covers a similar case of someone maliciously deleting photos, resulting in them being permanently deleted. 

The feature detailed below is described in part by other discussions.
- #2451 
- #2598 

[Feature] 
When a photo is delete in Immich it goes to the trash can (I am going to assume the trash can is enabled, but this feature works either way). Then after xyz amount of time, the photo is removed from the trash and deleted permanently. This feature would change that behavior and, after xyz amount of time sitting in the trash can, it would be moved to a "Protected Trash" that is not easily accessible to the user. This would allow the user to recover these "deleted" photos in the event they were accidentally (or maliciously) deleted.

I mentioned that these photos in the "Protected Trash" are not easily accessible. So in order to access them (to view or modify), the user would need to do something to verify their identity. This could be done by authenticating with a second factor or also having a time delay (like 5 days) before they have access to this collection of photos. Then once that criteria is met, they can go in and restore a photo or permanently delete one.

So if there is a photo that someone really wants to delete, lets say an embarrassing photo, then it can be done. But to permanently delete a photo, it requires additional work of validating your identity.

This type of system seems counterintuitive, having essentially two trash cans. So I would think it would be an opt-in feature on a user-by-user basis. They could enable the feature in their settings and even specify the method for unlocking the "Protected Trash" (2nd factor, time, etc). This way one administrator doesn't have access to someone's "deleted" photos. But instead it is the user who owns those photos who has complete control. 

[Reasoning behind Feature] 
The reason why I'm suggesting this is because one of the family members who I host Immich for, they accidentally deleted several images but thought they were only deleting one. They were surprised several weeks later when they couldn't find a particular photo. I of course have backups, but it was very difficult to restore those photos while preserving the newer images that occurred since the photos were deleted. 

This made me think, there needs to be a feature that protects Immich users from themselves. Imagine someone, who doesn't often use computers or is in a hurry, accidentally deletes some really important memories. Or another situation, if someone gets access to your phone and then maliciously deletes all of the photos. These memories are not replaceable and I think there needs to be better protection beyond a full system backup.

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by nicomerl** [comment]
Honestly, there IS a feature for This: trashbin. You can adjust the time after Pictures will be deleted.

I think making the trashbin more complex and impelenting more Features doesnt make sense. You can just restore missing pictures from your Backup in the worst case.

**Comment by bo0tzz** [comment]
I agree with @nicomerl's comment. This isn't something any other application does, and it just overcomplicates things.
